From fff9effb502bd37c01aa637dbd6f864bf23b3d24 Mon Sep 17 00:00:00 2001
From: Larisa Grigore <larisa.grigore@nxp.com>
Date: Fri, 22 Jan 2021 14:49:21 +0200
Subject: [PATCH 60/76] s32g_emu: Remove Linflex clock initialization

Remove the clock initialization since it makes
uart print very slow.

Issue: ALB-6137
Signed-off-by: Larisa Grigore <larisa.grigore@nxp.com>
---
 drivers/clk/s32/early_clocks.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/clk/s32/early_clocks.c b/drivers/clk/s32/early_clocks.c
index 72e9e982f3..da6e70d190 100644
--- a/drivers/clk/s32/early_clocks.c
+++ b/drivers/clk/s32/early_clocks.c
@@ -53,11 +53,13 @@ static struct clk mc_cgm0_mux0 = CLK_INIT(S32GEN1_CLK_MC_CGM0_MUX0);
 static struct clk xbar_2x = CLK_INIT(S32GEN1_CLK_XBAR_2X);
 
 /* LINFLEX clock */
+#ifndef CONFIG_TARGET_TYPE_S32GEN1_EMULATOR
 static struct clk periph_pll_mux = CLK_INIT(S32GEN1_CLK_PERIPH_PLL_MUX);
 static struct clk periph_pll_vco = CLK_INIT(S32GEN1_CLK_PERIPH_PLL_VCO);
 static struct clk periph_pll_phi3 = CLK_INIT(S32GEN1_CLK_PERIPH_PLL_PHI3);
 static struct clk mc_cgm0_mux8 = CLK_INIT(S32GEN1_CLK_MC_CGM0_MUX8);
 static struct clk lin_baud = CLK_INIT(S32GEN1_CLK_LIN_BAUD);
+#endif
 
 /* DDR clock */
 static struct clk ddr_pll_mux = CLK_INIT(S32GEN1_CLK_DDR_PLL_MUX);
@@ -123,6 +125,7 @@ static int enable_xbar_clock(void)
 	return s32gen1_enable(&xbar_2x);
 }
 
+#ifndef CONFIG_TARGET_TYPE_S32GEN1_EMULATOR
 static int enable_lin_clock(void)
 {
 	int ret;
@@ -147,6 +150,7 @@ static int enable_lin_clock(void)
 
 	return s32gen1_enable(&lin_baud);
 }
+#endif
 
 static int enable_ddr_clock(void)
 {
@@ -201,9 +205,14 @@ int enable_early_clocks(void)
 	if (ret)
 		return ret;
 
+#ifndef CONFIG_TARGET_TYPE_S32GEN1_EMULATOR
+	/* On emulator, changing the LIN clock frequency will
+	 * make the print very slow.
+	 */
 	ret = enable_lin_clock();
 	if (ret)
 		return ret;
+#endif
 
 	return enable_ddr_clock();
 }
-- 
2.17.1

